ocp_resources/datavolume.py

```diff
@@ -229,3 +244,36 @@ def _check_none_pending_status(self, failure_timeout=120):
         except TimeoutExpiredError:
             self.logger.error(f"{self.name} status is {sample}")
             raise
+
+    def is_garbage_collector_enabled_dv(self):
+        if self.exists:
+            dv_annotations = self.instance.metadata.get("annotations")
+            if dv_annotations:
+                return (
+                    dv_annotations.get("cdi.kubevirt.io/storage.deleteAfterCompletion")
+                    is not False
+                )
+        return True
+
+    def wait_for_dv_success(self, timeout=600, failure_timeout=120):
+        LOGGER.info(f"Wait DV success for {timeout} seconds")
+        self._check_none_pending_status(failure_timeout=failure_timeout)
+        self.pvc.wait_for_status(
+            status=PersistentVolumeClaim.Status.BOUND, timeout=timeout
+        )
+        # if garbage collector enabled, the DV deleted once succeeded
+        if (
+            is_garbage_collector_enabled_cdi_config()
+            and self.is_garbage_collector_enabled_dv()
+        ):
+            super().wait_deleted(timeout=timeout)
```

Comment:
1. Isn't `timeout` affected by GC `dataVolumeTTLSeconds`? what if it is set to be higher than `timeout`?
2. `wait_deleted` with 600s timeout is too high
3. iiuc the flow, you should first do
`self.wait_for_status(status=self.Status.SUCCEEDED, timeout=timeout)` and if GC is enabled, call `wait_deletd`